projects/components/src/time-range/time-range-for-page/time-range-for-page.component.ts

```diff
@@ -0,0 +1,37 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { ActivatedRoute, UrlSegment } from '@angular/router';
+import { NavigationService, TimeRange, TimeRangeForPageService } from '@hypertrace/common';
+import { isNil } from 'lodash-es';
+
+@Component({
+  selector: 'ht-time-range-for-page',
+  template: ` <ht-time-range (timeRangeSelected)="this.onTimeRangeSelected($event)"></ht-time-range> `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TimeRangeForPageComponent {
+  public constructor(
+    private readonly timeRangeForPageService: TimeRangeForPageService,
+    private readonly navigationService: NavigationService
+  ) {}
+
+  public onTimeRangeSelected(selectedTimeRange: TimeRange): void {
+    const activatedRoute = this.navigationService.getCurrentActivatedRoute();
+    const urlSegments: UrlSegment[] = activatedRoute.pathFromRoot.flatMap(activeRoute => activeRoute.snapshot.url);
+
+    if (urlSegments.length === 1 && this.shouldSavePageTimeRange(activatedRoute)) {
+      this.savePageTimeRange(selectedTimeRange, urlSegments[0]);
```

Comment:
@anandtiwary This is how it's determining if a time range should be saved or not (only left nav pages is currently the requirement), but i don't think this is great. For consumers that don't have `defaultTimeRange` set on their routes, page level time range functionality will be broken.

What is a better qualifier ? Is it enough to say, if there's one current urlSegment(indicating first level page) then save?